deploy/custom_resources/custom_authorizer/jwt_services.py

```diff
@@ -53,49 +20,55 @@ def fetch_public_keys():
     'verify_iss': True,
     'verify_sub': True,
     'verify_jti': True,
-    'verify_at_hash': False,
-    'require_aud': True,
-    'require_iat': True,
-    'require_exp': True,
-    'require_nbf': False,
-    'require_iss': True,
-    'require_sub': True,
-    'require_jti': True,
-    'require_at_hash': False,
-    'leeway': 0,
+    'verify_at_hash': True,
+    'require': ['aud', 'iat', 'exp', 'iss', 'sub', 'jti'],  # "nbf", "at_hash"
 }
 
 
 class JWTServices:
```

Comment:
I would convert the static functions to member methods and make this class instantiable.
Then in `__init__`... 
* I would `_fetch_openid_url` and store it in an attribute
* I would instantiate the PyJWKClient (which I assume is fetching the public key)

Then in the lambda handler I would instantiate the JWTServices() as a global variable to allow me to reuse it (which will avoid 2 network requests per invocation when the lambda is hot)